packages/common/src/DeployedModule.ts

```diff
@@ -0,0 +1,115 @@
+import {
+    AccountAddress,
+    AccountSigner,
+    AccountTransactionType,
+    CcdAmount,
+    ConcordiumGRPCClient,
+    ContractTransactionMetadata,
+    HexString,
+    InitContractPayload,
+    TransactionExpiry,
+    VersionedModuleSource,
+    getContractUpdateDefaultExpiryDate,
+    signTransaction,
+} from '.';
+import { checkParameterLength } from './contractHelpers';
+import { ModuleReference } from './types/moduleReference';
+import { Buffer } from 'buffer/';
+
+/**
+ * An update transaction without header.
+ */
+export type ContractInitTransaction = {
+    /** The type of the transaction, which will always be of type {@link AccountTransactionType.InitContract} */
+    type: AccountTransactionType.InitContract;
+    /** The payload of the transaction, which will always be of type {@link InitContractPayload} */
+    payload: InitContractPayload;
+};
+
+/** Type representing a smart contract module deployed on chain. */
+export class DeployedModule<C extends string = string> {
+    /** Private constructor to enforce creating objects using a static method. */
+    private constructor(
+        /** The gRPC connection used by this object */
+        public grpcClient: ConcordiumGRPCClient,
+        /** The reference for this module */
+        public moduleReference: ModuleReference
+    ) {}
+
+    /** Create a new `GenericModule` instance for interacting with a smart contract module on chain.
+     * This function ensures the module is already deployed on chain otherwise produces an error. */
+    public static async create(
+        grpcClient: ConcordiumGRPCClient,
+        moduleReference: ModuleReference
+    ): Promise<DeployedModule> {
+        const mod = new DeployedModule(grpcClient, moduleReference);
+        await mod.checkOnChain();
+        return mod;
+    }
```

Comment:
```suggestion
    public static async create<C extends string = string>(
        grpcClient: ConcordiumGRPCClient,
        moduleReference: ModuleReference
    ): Promise<DeployedModule<C>> {
        const mod = new DeployedModule<C>(grpcClient, moduleReference);
        await mod.checkOnChain();
        return mod;
    }
```

+ `@template C` declaration